Edit ingest management troubleshooting docs (#1160)
elastic/stack-docs: docs/en/ingest-management/troubleshooting.asciidoc

experimental[]

This early release of {ingest-management} is experimental. We invite you to
install and test these capabilities **in a test environment**. You might run
into problems and need to modify your setup to get this feature running.

We have collected the most common known problems and frequently asked questions
here. If your question isn't answered here, please review open issues in the
following GitHub repositories:

* https://github.com/elastic/kibana/issues[{kib}]
* https://github.com/elastic/beats/issuess[{beats}]
* https://github.com/elastic/package-registry/issues[{package-registry}]

> Contact us https://ela.st/ingest-manager-feedback[here]. Your feedback is very

Author: ```suggestion
Contact us in the https://ela.st/ingest-manager-feedback[discuss forum]. Your feedback is very
```

valuable to us.

> //REVIEWERS: Using "here" for link text is strongly discouraged because it's
> //bad for SEO. What text should we provide instead? 

Author: ```suggestion
```

**Common problems:**

* <<ingest-manager-not-in-kibana>>
* <<ingest-management-setup-fails>>
* <<ingest-manager-app-crashes>>
* <<agent-enrollment-timeout>>

**Frequently asked questions:**

* <<enrolled-agent-not-showing-up>>
* <<where-are-the-agent-logs>>
* <<what-is-my-agent-config>>
* <<where-is-the-data-agent-is-sending>>
* <<i-deleted-my-agent>>
* <<i-need-to-stop-agent>>
* <<i-rebooted-my-host>>
* <<what-is-the-endpoint-package>>

[float]
[[ingest-manager-not-in-kibana]]
== The {ingest-manager} app is not listed in the {kib} side navigation

In 7.8, the {ingest-manager} app is experimental. You must enable the app to
see it in {kib}.

> //REVIEWERS: I don't want to have separate sections for cloud vs self-managed
> //because we are working on a widget for switching between instructions for
> //the two. Also I think it's more confusing to users if we have two sections
> //that cover the same content. 
> 

Author: ```suggestion
```

//TODO: Add platform tabs when the tabbed panel widget is stable (possibly after
// 7.8)

**To enable {ingest-manager} on cloud:**

. Go to your deployment in the user console.

. Under the deployment name in the side navigation, click **Edit**.

. In the {kib} section, expand **User setting overrides** and enter the
following setting:
+
[source,yaml]
----
xpack.ingestManager.enabled: true
----

. Click **Save**.

{kib} will restart automatically. When {kib} is available, refresh the browser
to see the {ingest-manager} app in the navigation menu.

**To enable {ingest-manager} on a self-managed cluster:**

> //REVIEWERS: Can't on-premise also mean cloud? I've seen self-managed used, too
> //so I've used that here. 
> 

Author: ```suggestion
```

experimental[]

This early release of {ingest-management} is experimental. We invite you to
install and test these capabilities **in a test environment**. You might run
into problems and need to modify your setup to get this feature running.

We have collected the most common known problems and frequently asked questions
here. If your question isn't answered here, please review open issues in the
following GitHub repositories:

* https://github.com/elastic/kibana/issues[{kib}]
* https://github.com/elastic/beats/issuess[{beats}]
* https://github.com/elastic/package-registry/issues[{package-registry}]

Contact us https://ela.st/ingest-manager-feedback[here]. Your feedback is very
valuable to us.

//REVIEWERS: Using "here" for link text is strongly discouraged because it's
//bad for SEO. What text should we provide instead? 

> // REVIEWERS: Do you think it's worthwhile to separate the jump list into two 

Author: ```suggestion
```

> // sections? We could create separate topics with a jump list at the top, but
> // then users can't search the page. In general, presenting a list of more than
> // 8 options makes the info harder for users to process. 

Author: ```suggestion
```

> // REVIEWERS: Do you think it's worthwhile to separate the jump list into two 
> // sections? We could create separate topics with a jump list at the top, but
> // then users can't search the page. In general, presenting a list of more than
> // 8 options makes the info harder for users to process. 
> 

Author: ```suggestion
```

. Under the deployment name in the side navigation, click **Edit**.

. In the {kib} section, expand **User setting overrides** and enter the
following setting:
+
[source,yaml]
----
xpack.ingestManager.enabled: true
----

. Click **Save**.

{kib} will restart automatically. When {kib} is available, refresh the browser
to see the {ingest-manager} app in the navigation menu.

**To enable {ingest-manager} on a self-managed cluster:**

//REVIEWERS: Can't on-premise also mean cloud? I've seen self-managed used, too
//so I've used that here. 

. In the {es} configuration file, `config/elasticsearch.yml`, set the following
security settings to enable security and API keys:
+
[source,yaml]
----
xpack.security.enabled: true
xpack.security.authc.api_key.enabled: true
----

. In the {kib} configuration file, `config/kibana.yml`, enable {ingest-manager}
and specify user credentials:
+
[source,yaml]
----
xpack.ingestManager.enabled: true
xpack.ingestManager.fleet.tlsCheckDisabled: true <1>
xpack.security.enabled: true
elasticsearch.username: "elastic" <2>
elasticsearch.password: "abc123iUnbRftkABC123"
----
<1> This setting is not required if you configure TLS checking.

> <2> Specify a user who is authorized to use {ingest-manager} 

Reviewer: specifically the user needs to have 'superuser' privilege.  which is a little hectic, and only for the Alpha release (we hope).

Author: Yah, I was conflicted about whether to cover it here because we mention that elsewhere.

> <2> Specify a user who is authorized to use {ingest-manager} 

Author: ```suggestion
<2> Specify a user who is authorized to use {ingest-manager}.
```

Although it's redundant to install stand-alone {metricbeat}, you might want to
try installing it to see if it's able to send data successfully to {es}. For
more information, see
{metricbeat-ref}/metricbeat-getting-started.html[Get started with {metricbeat}].

If {metricbeat} is able to send data to {es}, there is possibly a bug or
problem with {agent}, and you should report it.

[float]
[[i-deleted-my-agent]]
== How do I restore an {agent} that I deleted from {fleet}?

It's ok, we've got your back! The data is still in {es}. To add {agent}
to {fleet} again, stop the agent, re-enroll the {agent} on the host, then run
{agent}.

[float]
[[i-need-to-stop-agent]]
== How do I stop {agent} and all processes running on my host?

To stop {agent} and its related executables, stop the {agent} process.

On Windows, if you installed the agent as a service, stop the service. If
necessary, use Task Manager on Windows to stop {agent}. This will kill the
{agent} process and any sub-processes it created (such as {beats}).

On Linux or macOS, run the following command to get the ID of the
`elastic-agent` process:

[source,shell]
----
ps | grep elastic-agent
----

Then kill the process.

> //REVIEWERS: Do we want to show the kill command. Not sure...if we do, which
> //kill command should we show?  kill -9? kill SIGTERM?

Author: ```suggestion
```

[float]
[[where-are-the-agent-logs]]
== Where does {agent} store logs after startup?

When started successfully, {metricbeat} logs are stored in
`data/logs/metricbeat` under the folder where {agent} was started. If that log
path does not exist, the agent was unable to start {metricbeat}, which is a
higher level problem to triage.

[float]
[[what-is-my-agent-config]]
== What configuration is the {agent} running?

To find the configuration file, inspect the `elastic-agent.yml` file in the
folder where you ran {agent}. If you're running the agent in {fleet} mode, this
file contains the following citation:

[source,yaml]
----
Management: mode: "fleet"
----

The `action_store.yml` contains the entire, unencrypted configuration:

* To see the {es} location, look at `outputs:hosts`.
* To see the {agent} version, look at the download folder and zip filenames.

This file also shows the version of all packages used by the current
configuration.

[float]
[[where-is-the-data-agent-is-sending]]
== Why can't I see the data {agent} is sending?

If {elastic-agent} is set up and running, but you don't see data in {kib}:

> //REVIEWERS: Is there a better command that we should recommend to users here?
> //I think it's good to show an actual command because we could have a newbie
> //reading this topic.

Author: ```suggestion
```

I think I have {agent} set up and running and there are no errors on the agent logs and I still don't see data in {kib}. The recommended way to assess this is to go straight to {es} data queries, or maybe just use {kib} Discover app. Look for any data in the index your {agent} is sending to (metrics-* for example) and see if the name.host of your host machine shows. If not, then perhaps the data is getting trapped / blocked in the network or isn't sent off of the {agent} host correctly in the first place due to some Firewall or security reason. While it should be redundant, it is all the same potentially helpful to install the stand-alone Metricbeat (see the Elastic docs for information) to see if that works to send data into ES, which would indicate a bug or problem in the {agent} side.
If {elastic-agent} is set up and running, but you don't see data in {kib}:

[float]
[[i-deleted-my-agent]]
== I deleted {agent} in the {fleet} {ingest-manager} app how can I get it back?
//REVIEWERS: Is there a better command that we should recommend to users here?
//I think it's good to show an actual command because we could have a newbie
//reading this topic.

> Its ok, we got your back!  The data is still in {es}. And get the {agent} back in the UI, you will need only to stop, then re-enroll the {agent} on the host and run it again.

Reviewer: I was trying so hard to be Elastic and put some humor in, I'm ok if you want to leave it out tho. Minimal sad panda tho.  About the command, its the same command ./elastic-agent run that is listed above we can re-cite it (or put it to a variable if its going to be repeated more) but that detail is up to you.

Author: Oh I know. I kept the comment...it's just a bit lower down. We just want to be a little careful about humor in troubleshooting topics. I think it works, tho, so let's go with it.

Author: About the comment...I was referring to the API call, not a command. My bad. I'll leave it as-is since you didn't flag it. Thanks!

> . Make sure there are no errors in {kib}. If there are errors, resolve them.

Reviewer: actually a known issue in the product is that kibana is showing an error when an agent is shut down, tho it is not really an 'error' state.  we may not want to explicitly say this as it can confuse users?  The start up option they need to do to get it back is on the host tho.  as noted above.

Author: aha...I will delete the line. In that case, I don't think we should mention errors in the intro to this section either.

> . Make sure there are no errors in {kib}. If there are errors, resolve them.

Author: ```suggestion
```
